src/CoreWCF.Primitives/src/CoreWCF/DXD.cs

```diff
@@ -0,0 +1,652 @@
+namespace CoreWCF
+{
+    using System.Xml;
+    using System.Collections.Generic;
+
+    // NOTE: This is a dynamic dictionary of XmlDictionaryStrings for the Binary Encoder to dynamically encode should
+    // the string not exist in the static cache.
+    // When adding or removing memebers please keep the capacity of the XmlDictionary field current.
+    static class DXD
+    {
+        static AtomicTransactionExternal11Dictionary atomicTransactionExternal11Dictionary;
+        static CoordinationExternal11Dictionary coordinationExternal11Dictionary;
+        static SecureConversationDec2005Dictionary secureConversationDec2005Dictionary;
+        static SecurityAlgorithmDec2005Dictionary securityAlgorithmDec2005Dictionary;
+        static TrustDec2005Dictionary trustDec2005Dictionary;
+        static Wsrm11Dictionary wsrm11Dictionary;
+
+        static DXD()
+        {
+            // Each string added to the XmlDictionary will keep a reference to the XmlDictionary so this class does
+            // not need to keep a reference.
+            XmlDictionary dictionary = new XmlDictionary(137);
+
+            // Each dictionaries' constructor should add strings to the XmlDictionary.
+            atomicTransactionExternal11Dictionary = new AtomicTransactionExternal11Dictionary(dictionary);
+            coordinationExternal11Dictionary = new CoordinationExternal11Dictionary(dictionary);
+            secureConversationDec2005Dictionary = new SecureConversationDec2005Dictionary(dictionary);
+            secureConversationDec2005Dictionary.PopulateSecureConversationDec2005();
+            securityAlgorithmDec2005Dictionary = new SecurityAlgorithmDec2005Dictionary(dictionary);
+            securityAlgorithmDec2005Dictionary.PopulateSecurityAlgorithmDictionaryString();
+            trustDec2005Dictionary = new TrustDec2005Dictionary(dictionary);
+            trustDec2005Dictionary.PopulateDec2005DictionaryStrings();
+            trustDec2005Dictionary.PopulateFeb2005DictionaryString();
+            wsrm11Dictionary = new Wsrm11Dictionary(dictionary);
+        }
+
+        static public AtomicTransactionExternal11Dictionary AtomicTransactionExternal11Dictionary
+        {
```

Comment:
I know this was wrong in the original source, but the order of the method modifiers is incorrect. Should be public static and not static public.